sql/catalyst/src/main/scala/org/apache/spark/sql/connector/expressions/expressions.scala

```diff
@@ -388,12 +388,13 @@ private[sql] object HoursTransform {
 }
 
 private[sql] final case class LiteralValue[T](value: T, dataType: DataType) extends Literal[T] {
-  override def toString: String = {
-    if (dataType.isInstanceOf[StringType]) {
-      s"'${StringUtils.replace(s"$value", "'", "''")}'"
-    } else {
-      s"$value"
-    }
+  override def toString: String = dataType match {
+    case StringType => s"'${StringUtils.replace(s"$value", "'", "''")}'"
+    case BinaryType =>
+      assert(value.isInstanceOf[Array[Byte]])
+      val bytes = value.asInstanceOf[Array[Byte]]
+      bytes.map("%02X".format(_)).mkString("X'", "", "'")
```

Comment:
Seems like `X` literal isn't universal and depends on dialect (https://github.com/apache/spark/pull/49452)

Reply:
Yes. you said right. The output information is just a generic represent like the others.

Reply:
How about to use `ApacheHex.encodeHexString(binary, ...` as in other places in the file.

Reply:
where else do we use `encodeHexString`?

Reply:
Spark SQL already parsed the format `X'123456'`.

The binary literal show the format `0x123456`.
```
  override def toString: String = value match {
     case null => "null"
     case binary: Array[Byte] => "0x" + ApacheHex.encodeHexString(binary, false)
```

Which is better?